### PR TITLE
Update faq to include dask FAQ

### PR DESCRIPTION
**Pull request by mgrover1**

### CODE REVIEW

File: faq.md [hunk 1]
```diff
@@ -1,2 +1,61 @@
...
+        # Name of the dimension to concatenate along.
+        concat_dim="time",
+
+        # Attempt to auto-magically combine the given datasets into one by using dimension coordinates.
+        combine="by_coords",
+
+        # Specify chunks for dask - explained later
+        chunks={"lev": 1, "time": 500},
+
+        # Only data variables in which the dimension already appears are included.
```
Comment (dcherian) [inline faq.md:27]:
```suggestion
        # Only data variables in which the concat_dim already appears are concatenated.
```

File: faq.md [hunk 2]
```diff
@@ -1,2 +1,61 @@
...
+        data_vars="minimal",
+
+        # Only coordinates in which the dimension already appears are included.
```
Comment (dcherian) [inline faq.md:30]:
```suggestion
        # Only non-dim coordinate variables in which concat_dim already appears are concatenated.
```

File: faq.md [hunk 3]
```diff
@@ -1,2 +1,61 @@
...
+        coords="minimal",
+
+        # Skip comparing and pick variable from first dataset.
```
Comment (dcherian) [inline faq.md:33]:
```suggestion
        # Skip comparing and pick variable from first dataset when concat_dim is not a dimension of a variable
```

File: faq.md [hunk 4]
```diff
@@ -1,2 +1,61 @@
...
+ds = xr.open_dataset(file, data_vars=['temperature'])
+```
+
+1. Your chunk size, using Dask, is not optimized. When reading a file in using Xarray with Dask, a "general rule of thumb" is to keep your chunk size down to around 100 mb.
+
+For example, let's say you trying to read in multiple files, each with ~600 time steps. This is case where each file is very large (several 10s of GB) and using Dask to help with data processing is *essential*.
+
+You can check the size of each chunk by subsetting a single DataArray (ex. `ds['temperature']`)
+
+If you have very large chunks, try modifying the number of chunks you specify within  `xr.open_mfdataset(files, ..., chunks={'lev':1, "time": 500})` where lev and time are vertical and time dimensions respectively.
```
Comment (dcherian) [inline faq.md:53]:
```suggestion
If you have very large chunks, try modifying the number of chunks you specify within  `xr.open_mfdataset(files, ..., chunks={'lev':1, "time": 500})` where `lev` and `time` are dimension names.
```

File: faq.md [hunk 5]
```diff
@@ -1,2 +1,61 @@
...
+        # Only coordinates in which the dimension already appears are included.
+        coords="minimal",
+
+        # Skip comparing and pick variable from first dataset.
+        compat="override",
+        parallel=True,
+    )
+```
+
+### My file is taking forever to read in... what can I do to improve this?
```
Comment (dcherian) [inline faq.md:39]:
```suggestion
### My file is taking forever to read in... what can I do to improve this?

1. Have you looked at [Xarray's multi-file documentation](https://xarray.pydata.org/en/stable/io.html#reading-multi-file-datasets)? Usually `data_vars="minimal", coords="minimal", compat="override"` works for model datasets that are consistent across the files you are reading in.
```

File: faq.md [hunk 6]
```diff
@@ -1,2 +1,61 @@
...
+
+If you have very large chunks, try modifying the number of chunks you specify within  `xr.open_mfdataset(files, ..., chunks={'lev':1, "time": 500})` where lev and time are vertical and time dimensions respectively.
+
+Check to see how large each chunk is after modifying the chunk size, and modify as necessary.
+
+1. You do not have enough dask workers
+
+If you have a few large files, having the number of workers equal to to the number of input files read in using `xr.open_mfdataset` would be a good practice
+
+If you have a large number of smaller files, you may not run into this issue, and it is suggest you look at the other potential solutions.
```
Comment (dcherian) [inline faq.md:61]:
```suggestion
If you have a large number of smaller files, you may not run into this issue, and it is suggest you look at the other potential solutions. If you are reading a large number of netCDF files, you should check the number of threads vs number of processes in your cluster. File reading is parallelized across processes, not threads so this is a potential bottleneck.
```

File: faq.md [hunk 7]
```diff
@@ -1,2 +1,134 @@
...
+1. xskillscore
+
+## Where can I find Xarray tutorials?
+
+See [videos](https://xarray.pydata.org/en/latest/tutorials-and-videos.html) and [notebooks](https://xarray-contrib.github.io/xarray-tutorial/).
+
+=======
+## How do I debug my code when using dask?
+>>>>>>> main
+
```
Comment (andersy005) [inline faq.md:41]: Bad merge?

**Pull request closed without merging**